atd-vzd/migrations/default/1697836965149_add_units_mode_category_function/up.sql

```diff
@@ -0,0 +1,59 @@
+CREATE OR REPLACE FUNCTION public.units_mode_category()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+DECLARE 
+    cris_unit_desc_id int;
+    cris_veh_body_styl_id int;
+BEGIN
+    -- Set the veh_body_styl_id and unit_desc_id variables from cris_facts schema on INSERT or
+    -- updated vz_facts table row on UPDATE
+    if  (TG_OP = 'INSERT') then
+        SELECT unit_desc_id, veh_body_styl_id INTO cris_unit_desc_id, cris_veh_body_styl_id 
+        FROM cris_facts.atd_txdot_units WHERE unit_nbr = NEW.unit_nbr AND crash_id = NEW.crash_id;
+    end if;
+
+    if  (TG_OP = 'UPDATE') then
+        cris_unit_desc_id := NEW.unit_desc_id;
+        cris_veh_body_styl_id := NEW.veh_body_styl_id;

```

Comment:
if there have been no VZ edits to `unit_desc_id` or `veh_body_styl_id` then these values would just be `null` right? and then the `atd_mode_category` wouldn't be set correctly? am i understanding this correctly. seems like in some cases this would still need to use the values from the cris schema table.

Reply:
👀 Thanks for catching this! You are right that I need to consider the lack of data in the `vz_facts` table. I'll do some thinking on this. 🙏